svgdigitizer/electrochemistry/electrolyte.py

```diff
@@ -0,0 +1,197 @@
+
+    def estimate_pH(self):
+        r"""
+        Estimates the pH of aquoues electrolytes.
+
+        EXAMPLES::
+
+            CID can be retrieved with InChIKey::
+            >>> from svgdigitizer.electrochemistry.electrolyte import Electrolyte
+            >>> electrolyte_dict = {"type": "aq", "components": [{"name": "H2SO4","type": "acid", "concentration": {"value": 0.1, "unit": "M"}}, {"name": "Na2SO4","type": "salt", "concentration": {"value": 0.1, "unit": "M"}}]}
+            >>> electrolyte = Electrolyte(electrolyte_dict)
+            >>> electrolyte.pH
+            1.3538681030273414
+        """
+
+        if self.electrolyte['type'] != 'aq':
+            raise NotImplementedError("pH calculations are only implemented for aqueous solutions!")
+        acids_bases = []
+        for i in self.electrolyte['components']:
+            if not i['type'] == 'solvent':
+                InChIKey = self.lookup_chemical_ids_from_name(i['name'])["InChIKey"]
+                temp = self.get_pKa(InChIKey)
+                for e in temp:
+                    q = i['concentration']['value'] * normalize_unit(i['concentration']['unit'])
+                    e.conc = q.to("mol / l").value
+                acids_bases.extend(temp)
+
+        system = System(*acids_bases)
+        system.pHsolve()
+
+        return system.pH
+
+    @classmethod
+    def lookup_CID_from_InChI(cls, InChIKey=None, InChI=None):
+        r"""
+        Lookup Pubchem CID from InChI or InChiKey.
+
+        EXAMPLES::
+
+            CID can be retrieved with InChIKey::
+            >>> from svgdigitizer.electrochemistry.electrolyte import Electrolyte
+            >>> Electrolyte.lookup_CID_from_InChI(InChIKey='NBIIXXVUZAFLBC-UHFFFAOYSA-N') # phosphoric acid
+            1004
+
+            CID can be retrieved with InChI::
+            >>> from svgdigitizer.electrochemistry.electrolyte import Electrolyte
+            >>> Electrolyte.lookup_CID_from_InChI(InChI='InChI=1S/H3O4P/c1-5(2,3)4/h(H3,1,2,3,4)') # phosphoric acid
+            1004
+
+        """
+        if InChIKey:
+            return pcp.get_cids(InChIKey, 'inchikey')[0]
+        elif InChI:
+            return pcp.get_cids(InChI, 'inchi')[0]
+
+    @classmethod
+    def lookup_chemical_ids_from_name(cls, chemical_name):
+        r"""
+        Lookup CID, InChI, and InChIKey from chemical name.
+
+        EXAMPLES::
+
+            CID can be retrieved with InChIKey::
+            >>> from svgdigitizer.electrochemistry.electrolyte import Electrolyte
+            >>> Electrolyte.lookup_chemical_ids_from_name('phosphoric acid')
+            {'CID': 1004, 'InChI': 'InChI=1S/H3O4P/c1-5(2,3)4/h(H3,1,2,3,4)', 'InChIKey': 'NBIIXXVUZAFLBC-UHFFFAOYSA-N'}
+
+        """
+        results = pcp.get_cids(chemical_name, 'name')
+        len_results = len(results)
+        if len_results > 1:
+            raise RuntimeError("Ambiguous chemical name! More than one chemical id return by pubchem. Use another chemical name")
```

Comment:
```suggestion
            raise RuntimeError("Ambiguous chemical name! More than one chemical id returned by pubchem. Use another chemical name")
```

Reply:
Maybe it would be helpful if the result is printed, to understand where the error might be.
Not sure if `results` prints nicely.

Reply:
```results``` is only a list of pubchem CIDs. Printing something like a list of synonyms for all CIDs would require more requests to pubchem.